source/postcard-schema/src/lib.rs

```diff
@@ -14,4 +16,25 @@ pub trait Schema {
     /// A recursive data structure that describes the schema of the given
     /// type.
     const SCHEMA: &'static schema::NamedType;
+
+    /// A manually calculated maximum size hint.
+    ///
+    /// This is useful in the case that a foreign type is expressed
+    /// in some form that has a technically unbounded max size, for
+    /// example a byte slice, but in reality it never sends more than
+    /// some calcuable size.
+    ///
+    /// A good example is `heapless::Vec<T, N>`, which is expressed
+    /// on the wire as a `Seq(N)`, which is unbounded, however the
+    /// true max size is the length to store N as a varint plus
+    /// `N * max_size::<T>`, which is calcuable.
+    ///
+    /// This value should not typically be used directly, instead use
+    /// [`max_size()`][crate::max_size::max_size()], which will EITHER
+    /// use the manual max size, or calculate the max size from the
+    /// schema.
+    ///
+    /// You must not rely on this value for safety reasons, as implementations
+    /// could be wrong.
+    const MANUAL_MAX_SIZE: Option<usize> = None;
```

Comment:
Instead of having this manual override and a `max_size()` function that checks it, this could default to the calculated size and be overridden as appropriate:

```rust
trait Schema {
    const MAX_SIZE: Option<usize> = max_size_nt(Self::SCHEMA);
}

// Manual max size
impl Schema for heapless::Vec<T, N> {
    const MAX_SIZE: Option<usize> = Some(...);
}
```

This seems more intuitive to me, since there'd be only one source of truth--`T::MAX_SIZE`--instead of the hint and the calculator function.